esitoolsupport/esibase.py

```diff
@@ -0,0 +1,122 @@
+# Copyright 2009-2015 Eucalyptus Systems, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; version 3 of the License.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see http://www.gnu.org/licenses/.
+#
+# Please contact Eucalyptus Systems, Inc., 6755 Hollister Ave., Goleta
+# CA 93117, USA or visit http://www.eucalyptus.com/licenses/ if you need
+# additional information or have any questions.
+
+#
+# Order matters here. We want to make sure we initialize logging before anything
+# else happens. We need to initialize the logger that boto will be using.
+#
+
+import sys
+import subprocess
+import os
+from urlparse import urlparse
+from boto.sts import STSConnection
+
+class EsiBase(object):
+
+    def __init__(self, region='localhost'):
+        self.vars = {}
+        self.region = region
+        self._load_vars()
+        self.check_environment()
+        self._set_environment()
+
+    def get_sts_connection(self):
+        token_url = urlparse(self.vars['TOKEN_URL'])
+        STSConnection.DefaultRegionEndpoint = token_url.hostname
+        port = token_url.port if token_url.port else 80
+        return STSConnection(is_secure=False, port=port, path=token_url.path,
+                                aws_access_key_id=self.get_env_var('AWS_ACCESS_KEY_ID'),
+                                aws_secret_access_key=self.get_env_var('AWS_SECRET_ACCESS_KEY'))
+
+    def _set_environment(self):
+        for i in ("AWS_ACCESS_KEY_ID", "AWS_SECRET_ACCESS_KEY"):
+            os.environ[i] = self.vars[i]
+
+    def list_system_accounts(self):
+        accounts = {}
+        process = subprocess.Popen(['/usr/bin/euare-accountlist', '-U', self.vars['AWS_IAM_URL']],
+                                   stdout=subprocess.PIPE)
+        while process.poll() is None:
+            try:
+                line = process.stdout.readline()
+                if line:
+                    a = line.strip().split('\t')
+                    if len(a) == 2 and a[0].startswith('(eucalyptus)'):
+                        accounts[a[0]] = a[1]
+            except:
+                pass
+        return accounts
```

Comment:
Don't overthink it too much.  Simply iterating over a file object will give you a line at a time until it hits EOF.  A more reasonable way to process euare-accountlist's output is as follows:

```python
process = subprocess.Popen(['euare-accountlist', '-I', self.vars['AWS_IAM_URL']],
                           stdout=subprocess.PIPE)
for line in process.stdout:
    split = line.strip().split(None, 1)
    if len(split) > 1 and split[0].startswith('(eucalyptus)'):
        accounts[split[0]] = split[1]
```